model/channel_sidebar.go

```diff
@@ -109,3 +110,13 @@ func (o OrderedSidebarCategories) ToJson() []byte {
 		return b
 	}
 }
+
+func IsValidCategoryId(s string) bool {
+	// Category IDs can either be regular IDs
+	if IsValidId(s) {
+		return true
+	}
+
+	// Or default categories can follow the pattern {type}_{userID}_{teamID}
+	return regexp.MustCompile("(favorites|channels|direct_messages)_[a-z0-9]{26}_[a-z0-9]{26}").MatchString(s)
```

Comment:
I tried to avoid a regex here, but the extra underscore in direct_messages made it hard to split the string on an underscore and check each part individually.